lib/modules/manager/bazel/rules/index.ts

```diff
@@ -1,74 +1,44 @@
-import is from '@sindresorhus/is';
-import { logger } from '../../../../logger';
+import { ZodObject, z } from 'zod';
 import { regEx } from '../../../../util/regex';
 import type { PackageDependency } from '../../types';
-import type {
-  Fragment,
-  StringFragment,
-  Target,
-  TargetAttribute,
-} from '../types';
-import { dockerDependency } from './docker';
-import { gitDependency } from './git';
-import { goDependency } from './go';
-import { httpDependency } from './http';
-
-type DependencyExtractor = (_: Target) => PackageDependency | null;
-type DependencyExtractorRegistry = Record<string, DependencyExtractor>;
-
-const dependencyExtractorRegistry: DependencyExtractorRegistry = {
-  git_repository: gitDependency,
-  go_repository: goDependency,
-  http_archive: httpDependency,
-  http_file: httpDependency,
-  container_pull: dockerDependency,
-};
-
-const supportedRules = Object.keys(dependencyExtractorRegistry);
+import { extract } from '../parser';
+import type { Fragment, FragmentData, Target } from '../types';
+import { DockerTarget } from './docker';
+import { GitTarget } from './git';
+import { GoTarget } from './go';
+import { HttpTarget } from './http';
+
+const Target = z.union([DockerTarget, GitTarget, GoTarget, HttpTarget]);
+
+/**
+ * Infer all rule names supported by Renovate in order to speed up parsing
+ * by filtering out other syntactically correct rules we don't support yet.
+ */
+const supportedRules = Target.options.reduce<string[]>((res, targetSchema) => {
+  const schema = targetSchema._def.schema;
```

Comment:
this looks a bit hacky. 🤔

Reply:
Always can fallback to explicit string enumeration in case of problems

Reply:
can we have some simpler code instead?